test/hotspot/jtreg/compiler/vectorapi/TestRotateWithZero.java

```diff
@@ -0,0 +1,304 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+ package compiler.vectorapi;
+
+ import java.util.Random;
+ import jdk.incubator.vector.IntVector;
+ import jdk.incubator.vector.ShortVector;
+ import jdk.incubator.vector.VectorOperators;
+ import jdk.incubator.vector.VectorSpecies;
+ import jdk.test.lib.Asserts;
+ import jdk.test.lib.Utils;
+
+ /*
+  * @test
+  * @bug 8351627
+  * @summary C2 AArch64 ROR/ROL: assert((1 << ((T>>1)+3)) > shift) failed: Invalid Shift value
+  * @modules jdk.incubator.vector
+  * @library /test/lib
+  * @run main/othervm -XX:-TieredCompilation compiler.vectorapi.TestRotateWithZero
+  */
+public class TestRotateWithZero {
+    private static final int INVOCATIONS = 10000;
+    private static final int LENGTH = 2048;
+    private static final VectorSpecies<Integer> I_SPECIES = IntVector.SPECIES_PREFERRED;
+    private static final VectorSpecies<Short> S_SPECIES = ShortVector.SPECIES_PREFERRED;
+    private static final Random random = Utils.getRandomInstance();
+
+    private static int[] arr1;
+    private static int[] arr2;
+    private static int[] res;
+    private static short[] sarr1;
+    private static short[] sarr2;
+    private static short[] sres;
+
+    static {
+        arr1 = new int[LENGTH];
+        arr2 = new int[LENGTH];
+        res = new int[LENGTH];
+        sarr1 = new short[LENGTH];
+        sarr2 = new short[LENGTH];
+        sres = new short[LENGTH];
+
+        for (int i = 0; i < LENGTH; i++) {
+            arr1[i] = random.nextInt();
+            arr2[i] = 0;
+            sarr1[i] = (short)random.nextInt();
+            sarr2[i] = (short)0;
+        }
+    }
+
+    private static void rotateRightWithZero() {
+        for (int i = 0; i < LENGTH; i += I_SPECIES.length()) {
+            IntVector v = IntVector.fromArray(I_SPECIES, arr1, i);
+            v.lanewise(VectorOperators.ROR, 0).intoArray(res, i);
+        }
+    }
+
+    private static void rotateLeftWithZero() {
+        IntVector vzero = IntVector.zero(I_SPECIES);
```

Comment:
This local is unused.

Reply:
Thanks for your review! Good catch. I will remove it soon.